modules/frontend/v1/frontendv1pb/frontend.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+// Protobuf package should not be changed when moving around go packages
+// in order to not break backward compatibility.
+package frontend;
+
+option go_package = "modules/frontend/v1/frontendv1pb";
```

Comment:
Does this need the pathing `modules/....`?  Just comparing to Loki which also [forked the cortex frontend recently](  https://github.com/grafana/loki/blob/main/pkg/lokifrontend/frontend/v1/frontendv1pb/frontend.proto) and wondering if that is related to the discussion about not gen'ing these in Makefile.

Reply:
I think the path just needs to be a depiction of where the module is located from the perspective of the `protoc` command.  In the note I left in the `Makefile`, you can see that we're including two `-I` flags to use as import paths.  In the case of the public packages, we pull them out of our vendor directory.  Perhaps we should make this the full public path for the package instead of the local relative path.  But if I leave the `go_package` name as a simple string, such as `frontendv1pb` I get a message like the following.

```
❯ protoc -I vendor -I . --go_out=./ --go_opt=paths=source_relative --go-grpc_out=./ --go-grpc_opt=paths=source_relative modules/frontend/v1/frontendv1pb/frontend.proto
protoc-gen-go: invalid Go import path "frontendv1pb" for "modules/frontend/v1/frontendv1pb/frontend.proto"

The import path must contain at least one period ('.') or forward slash ('/') character.

See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.
```

This situation started happening maybe a year or 18 months ago when newer versions of `protoc` started requiring a more complete package name.  There are probably a few ways around it, like compiling an older version of `proto` or executing the build from a different directory.

One thing I'm not clear on after reviewing the Loki change you pointed out, is why there isn't a version header in the generated pb.gp.  In ours after generation we have the following: 

```
// Code generated by protoc-gen-go. DO NOT EDIT.
// versions:
//      protoc-gen-go v1.27.1
//      protoc        v3.19.2
// source: modules/frontend/v1/frontendv1pb/frontend.proto
```

This might be related to building with a new set of tooling, or perhaps there is a flag the Loki folks are using to skip including the header.  In either case, at first glance I wasn't able to figure out what proto version they were using to generate the pb.go files.

Reply:
I'll follow up with the Loki team to see if there is an easier change we can make here.

Reply:
This might be more complicated than I'd originally thought.  I wasn't aware of the gogo/protobuf project but it seems that both cortex and loki are using this to generate the pb.go files.  Should we also think about this?

Also, both cortex and loki are using a build image which is pinned to a specific package version of the gogo/protobuf package, but perhaps this is not compatible with newer protobufs.

https://github.com/cortexproject/cortex/blob/master/build-image/Dockerfile#L36
https://github.com/grafana/loki/blob/main/loki-build-image/Dockerfile#L75

I also I see the gogo/protobuf project is looking for new ownership, and hasn't had much contribution recently, so I don't know where we should stand here, since this seems like the most we'll get of a deprecation warning if new ownership doesn't arrive to update the project for newer protobufs.

https://github.com/gogo/protobuf

I'm also not clear the benefits of this over google/protobuf.

Thoughts on how to move forward?

Reply:
Doing the gogoproto thing now.